src/ucp/rma/flush.c

```diff
@@ -737,7 +743,8 @@ static ucs_status_t ucp_worker_fence_strong(ucp_worker_h worker)
 
     do {
         ucp_worker_progress(worker);
-    } while (request->flush_worker.comp_count > 1);
+    } while ((request->flush_worker.comp_count > 1) ||
+             !ucp_worker_flush_all_eps_scheduled(request, worker));
```

Comment:
do you think it is worth to swap conditions for better perf?

Reply:
so it means request->flush_worker.comp_count is not really checking flush completion?
maybe we can wait for flush request completion in a better way?

Reply:
discussed offline to make this function to reuse code from blocking flush